### ISSUE TITLE
filename sorting causes label-frame mismatch in napari label frames GUI

### ISSUE THREAD

**Issue by backyardbiomech** [DeepLabCut/DeepLabCut#2128]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

This isn't so much a bug as an artifact that might be useful for others.

In a project that has many iterations over 2 years, some frames were extracted with fewer digits than others from the same video. E.g. `img744.png` vs `img0777.png`. As a result `img744.png` is sorted after `img0777.png` in `CollectedData.DLC`. Since this project started back with an early maDLC release candidate, and is on iteration 6, lots of different DLC versions have extracted frames or outliers. `img744.png` was extracted on Feb 11, 2021, and `img0777.png` was extracted July 15, 2021 (probably the most up to date releases on those dates).

The napari-based window appears to load the points layer based on the sorting from `CollectedData` (very literal alpha-numerical sorting with 0777 before 744), but the images are sorted "properly". In other words, the image for `img744.png` is shown before the image for `img0777.png`, but the points for `img0777.png` are shown before the points for `img0777.png`. When the image files are 

```
img0235.png
img0293.png
img0430.png
img0652.png
img0777.png
img1063.png
img1188.png
img744.png
img953.png
```
This creates havoc. 

Interestingly, like napari, windows file explorer sorts the images correctly (it "knows" that img744 comes before img0777, even when sorted alpha-numerically).
Also running `check-labels` produces accurately labeled images, I'm assuming because that uses full file paths in `CollectedData` to pul the proper image. 

So, there is no problem for the project until I want to extract outliers from these older videos (or use the GUI to check some labels quickly). 

With 800+ frames labeled across 30+ videos (and the project is growing more complex now), this is not an easy fix.
I think, as a workaround, a script to find these images with fewer digits, rename the images, and edit `CollectedData...` is the only fix, unless napari can be told to change sorting of the points layer.

### Operating System

Windows 10

### DeepLabCut version

2.3.0

### DeepLabCut mode

multi animal

### Device type

gpu

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by jeylau** [comment]
I had started to write some code to fix the image names when I realized we'd also need to update the image paths in all data pickles and other h5 files, which I find scary. It'd be safer to do it on napari's side, though this implies changing how `imread` and `glob` sort paths. I still prefer the latter, with maybe a checkbox somewhere in the GUI to toggle natural sorting... What do you think @backyardbiomech @MMathisLab @AlexEMG?

**Comment by backyardbiomech** [comment]
@jeylau I guess that's the best bet.
I brute-force changed the file names and the paths in CollectedData to get it to work. I was about to train a new iteration anyway, so my previous model is protected; there wasn't much need to worry about backwards compatibility of the other data files in my case. 
If your napari suggestion is challenging or risky, you could include a utility that changes the file names with the express warning about what will no longer work because you aren't updating the filenames *everywhere*. e.g., "If you proceed, you will no longer be able to evaluate models trained previously...". 

Or instructions on how to quickly update the network with existing tools after the change so everything works again. For example, would it work to merge (or advance iteration), create training data, train from previous snapshot for some very small number of iterations at a very small learning rate? Would that leave you with essentially the same model but now based on a training dataset with all updated image names and thus able to be evaluated?

**Comment by jeylau** [comment]
@backyardbiomech I may have found a very simple solution https://github.com/DeepLabCut/napari-deeplabcut/commit/23bbb6fe80c01c9a66a168f6db8e37a85865ae0b
If you get the chance to test it, let me know if that fixes the issue. I tried it after removing the 0-padding from a couple of images, and it seems to work great!

**Comment by MMathisLab** [comment]
fixed in the latest deeplabcut-napari < so please upgrade :)
